lib/secret/secret.go

```diff
@@ -108,6 +108,14 @@ func (k Key) Open(ciphertext []byte) ([]byte, error) {
 		return nil, trace.Wrap(err)
 	}
 
+	// Attempting to call aesgcm.Open with a invalid nonce will cause it to panic.
+	// To make sure that doesn't happen even when we send invalid ciphertext
+	// (for example, for legacy secret package or attacker controlled data),
+	// reject invalid sized nonces.
+	if len(data.Nonce) != aesgcm.NonceSize() {
+		return nil, trace.BadParameter("bad nonce")
```

Comment:
bad nonce is not a very descriptive, can you come up with something more user-friendly?